### PR TITLE
[CAS-1274] Substitute `ChatDomain::createChannel`

### PR DESCRIPTION
**Pull request by devUndef1ned**
### 🎯 Goal

Subtitute usage of `ChatDomain::createChannel`

### 🛠 Implementation details

1. New extension `ChatClient::createChannel(channle: Channel)`
2. New `CreateChannelService` that encapsulates logic of creating channel

### 🧪 Testing

Run old UI sample and create channel and be sure nothing has changed.

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#android-team or #compose-chat-sdk-team) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- <del> [ ] New code is covered by unit tests </del>
- <del> [ ] Comparison screenshots added for visual changes </del>
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/extensions/ChatClient.kt [hunk 1]
```diff
@@ -18,9 +18,17 @@ import io.getstream.chat.android.client.models.User
...
 import io.getstream.chat.android.offline.ChatDomainImpl
+import io.getstream.chat.android.offline.channel.CreateChannelService
 import io.getstream.chat.android.offline.usecase.DownloadAttachment
 import io.getstream.chat.android.offline.utils.validateCid
 
+/**
+ * Returns the instance of [ChatDomainImpl] as cast of singleton [ChatDomain.instance] to the [ChatDomainImpl] class.
+ */
+private fun domainImpl(): ChatDomainImpl {
+    return domainImpl()
```
Comment (amitkma) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/extensions/ChatClient.kt:29]: This is incorrect. It causes StackOverFlow exception due to recursion.

Comment (amitkma) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/extensions/ChatClient.kt:29]: This is incorrect. It causes StackOverFlow exception due to recursion.

**Review by amitkma — CHANGES_REQUESTED** [review]
Review in progress.

File: DEPRECATIONS.md [hunk 2]
```diff
@@ -4,6 +4,7 @@ This document lists deprecated constructs in the SDK, with their expected time 
 
 | API / Feature | Deprecated (warning) | Deprecated (error) | Removed | Notes |
 | --- | --- | --- | --- | --- |
+| `ChatDomain.createChannel` | 2022.01.04<br/>4.26.0 |  2022.01.18<br/>⌛ | 2022.02.01 ⌛ | Use extension function `ChatClient.createChannel` instead |
```
Comment (adasiewiczr) [inline DEPRECATIONS.md:7]:
```suggestion
| `ChatDomain.createChannel` | 2022.01.21<br/>4.27.0 |  2022.02.04<br/>⌛ | 2022.02.12 ⌛ | Use extension function `ChatClient.createChannel` instead |
```

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomain.kt [hunk 3]
```diff
@@ -326,6 +326,13 @@ public sealed interface ChatDomain {
      *
      * @see io.getstream.chat.android.livedata.utils.RetryPolicy
      */
+    @Deprecated(
+        message = "createChannel is deprecated. Use extension function ChatClient::createChannel instead",
```
Comment (adasiewiczr) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomain.kt:330]:
```suggestion
        message = "createChannel is deprecated. Use ChatClient::createChannel extension function instead",
```

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/ChatDomain.kt [hunk 4]
```diff
@@ -332,6 +332,13 @@ public sealed interface ChatDomain {
      *
      * @see io.getstream.chat.android.offline.utils.RetryPolicy
      */
+    @Deprecated(
+        message = "createChannel is deprecated. Use extension function ChatClient::createChannel instead",
```
Comment (adasiewiczr) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/ChatDomain.kt:336]:
```suggestion
        message = "createChannel is deprecated. Use ChatClient::createChannel extension function  instead",
```

**Review by adasiewiczr — COMMENTED** [review]
@amitkma can you take a look on tests?

**Comment by amitkma** [comment]
I started to look into tests. Will update PR today.

**Review by amitkma — DISMISSED** [review]

**Review by amitkma — COMMENTED** [review]
Looks good to me. @adasiewiczr could you please review?

**Review by adasiewiczr — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [x] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Pull request merged**